### ISSUE TITLE
How to install on macOS?

### ISSUE THREAD

**Issue by jroutines** [johnBuffer/AntSimulator#6]
The README is too simple for newcomer.

**Comment by sisco0** [comment]
Hello @jroutines, did you try using `brew install make cmake sfml gcc` and, after that running the next commands:
```bash
cmake . -Bbuild
cmake --build build
```

Then, a binary `AntSimulator` file would appear at the build folder, which you could invoke by using:
`build/AntSimulator`

**Comment by duocang** [comment]
@jroutines 

1. Download Cmake and install cmake
2. Downloadn [SFML source](https://www.sfml-dev.org/files/SFML-2.5.1-sources.zip) and create a build folder in SFML folder (other place is ok too but it is normal to do what i said)
3. Open Cmake and do what I show you
4. Click Configure and Generate one by one (**Now, your SFML is ready to be built**)
   
![image](https://user-images.githubusercontent.com/14146871/113845858-35e4ae80-9796-11eb-83ca-708185f5a2e2.png)

5. open command line and go to your SFML build folder
![image](https://user-images.githubusercontent.com/14146871/113846433-befbe580-9796-11eb-8279-357678678b3f.png)

6. type in `make` and enter,  you should see things flying
![image](https://user-images.githubusercontent.com/14146871/113846640-fc607300-9796-11eb-855e-9504512bd319.png)

7. download AntSimulator and create a build folder in it
8. Cmake and Configure button
![image](https://user-images.githubusercontent.com/14146871/113847448-c96aaf00-9797-11eb-8d4c-d5489b822c3e.png)

9. After Configure button, tell Cmake where is your SFML
![image](https://user-images.githubusercontent.com/14146871/113847726-0a62c380-9798-11eb-8479-557d7bb91ca8.png)

10. go to build folder and make

now you have build everything.


![image](https://user-images.githubusercontent.com/14146871/113847894-34b48100-9798-11eb-831d-a5e68acb9c10.png)


If you donot know what I said above then you should go to learn cmd line and what is cmake.

**Comment by pancak3** [comment]
I assume this is not compatible with M1 mac?
Cuz I tried and it says `ld: symbol(s) not found for architecture arm64` when I run `make`.
It is weird since I already have sfml built from source.
Maybe I need to hack cmakefile.txt.

**Comment by duocang** [comment]
@pancak3 

It seems to me not an M1 problem. You better show your error message because ld: symbols missing is a common issue when you need some libs.

**Comment by pancak3** [comment]
@duocang thanks for help! The logs are as fellow,

```
....
....
1 warning generated.
[ 75%] Building CXX object CMakeFiles/AntSimulator.dir/src/utils.cpp.o
[100%] Linking CXX executable AntSimulator
ld: warning: ignoring file /usr/local/./sfml-graphics.framework/Versions/2.5.1/sfml-graphics, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /usr/local/./sfml-window.framework/Versions/2.5.1/sfml-window, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /usr/local/./sfml-system.framework/Versions/2.5.1/sfml-system, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "sf::CircleShape::CircleShape(float, unsigned long)", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
      Food::render(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
      Ant::render_food(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
  "sf::VertexArray::resize(unsigned long)", referenced from:
      World::World(unsigned int, unsigned int) in main.cpp.o
  "sf::VertexArray::VertexArray(sf::PrimitiveType, unsigned long)", referenced from:
      DisplayManager::DisplayManager(sf::RenderTarget&, sf::RenderWindow&, World&, Colony&) in display_manager.cpp.o
      World::World(unsigned int, unsigned int) in main.cpp.o
      Colony::Colony(float, float, unsigned int) in main.cpp.o
  "sf::VertexArray::operator[](unsigned long)", referenced from:
      World::renderMarkers(sf::RenderTarget&, sf::RenderStates) const in display_manager.cpp.o
      World::renderWalls(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
      Ant::render_in(sf::VertexArray&, unsigned int) const in display_manager.cpp.o
      World::World(unsigned int, unsigned int) in main.cpp.o
      Colony::Colony(float, float, unsigned int) in main.cpp.o
  "sf::RenderStates::RenderStates()", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
  "sf::RenderTarget::draw(sf::Drawable const&, sf::RenderStates const&)", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
      World::renderMarkers(sf::RenderTarget&, sf::RenderStates) const in display_manager.cpp.o
      World::renderWalls(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
      Food::render(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
      Ant::render_food(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
  "sf::RenderTarget::clear(sf::Color const&)", referenced from:
      _main in main.cpp.o
  "sf::RenderWindow::RenderWindow(sf::VideoMode, sf::String const&, unsigned int, sf::ContextSettings const&)", referenced from:
      _main in main.cpp.o
  "sf::RenderWindow::~RenderWindow()", referenced from:
      _main in main.cpp.o
  "sf::Transformable::setPosition(sf::Vector2<float> const&)", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
      Food::render(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
      Ant::render_food(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
  "sf::Transformable::setOrigin(float, float)", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
      Food::render(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
      Ant::render_food(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
  "sf::RectangleShape::RectangleShape(sf::Vector2<float> const&)", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
  "sf::Clock::Clock()", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
  "sf::Color::Black", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
  "sf::Color::Color(unsigned char, unsigned char, unsigned char, unsigned char)", referenced from:
      World::renderMarkers(sf::RenderTarget&, sf::RenderStates) const in display_manager.cpp.o
      World::renderWalls(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
      ___cxx_global_var_init in display_manager.cpp.o
      ___cxx_global_var_init.1 in display_manager.cpp.o
      ___cxx_global_var_init.2 in display_manager.cpp.o
      ___cxx_global_var_init.7 in display_manager.cpp.o
      _main in main.cpp.o
      ...
  "sf::Image::loadFromFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      _main in main.cpp.o
  "sf::Image::Image()", referenced from:
      _main in main.cpp.o
  "sf::Image::~Image()", referenced from:
      _main in main.cpp.o
  "sf::Mouse::getPosition(sf::Window const&)", referenced from:
      DisplayManager::processEvents() in display_manager.cpp.o
      _main in main.cpp.o
  "sf::Shape::setFillColor(sf::Color const&)", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
      Food::render(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
      Ant::render_food(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
  "sf::Shape::~Shape()", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
      Food::render(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
      Ant::render_food(sf::RenderTarget&, sf::RenderStates const&) const in display_manager.cpp.o
  "sf::String::String(char const*, std::__1::locale const&)", referenced from:
      _main in main.cpp.o
  "sf::Window::setFramerateLimit(unsigned int)", referenced from:
      DisplayManager::processEvents() in display_manager.cpp.o
      _main in main.cpp.o
  "sf::Window::close()", referenced from:
      DisplayManager::processEvents() in display_manager.cpp.o
  "sf::Window::display()", referenced from:
      _main in main.cpp.o
  "sf::Window::pollEvent(sf::Event&)", referenced from:
      DisplayManager::processEvents() in display_manager.cpp.o
  "sf::Texture::loadFromFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, sf::Rect<int> const&)", referenced from:
      DisplayManager::DisplayManager(sf::RenderTarget&, sf::RenderWindow&, World&, Colony&) in display_manager.cpp.o
      DefaultConf<int>::loadTextures() in main.cpp.o
  "sf::Texture::setSmooth(bool)", referenced from:
      DefaultConf<int>::loadTextures() in main.cpp.o
  "sf::Texture::Texture()", referenced from:
      DisplayManager::DisplayManager(sf::RenderTarget&, sf::RenderWindow&, World&, Colony&) in display_manager.cpp.o
      DefaultConf<int>::loadTextures() in main.cpp.o
  "sf::Texture::~Texture()", referenced from:
      DisplayManager::DisplayManager(sf::RenderTarget&, sf::RenderWindow&, World&, Colony&) in display_manager.cpp.o
      _main in main.cpp.o
      std::__1::__shared_ptr_emplace<sf::Texture, std::__1::allocator<sf::Texture> >::~__shared_ptr_emplace() in main.cpp.o
      std::__1::__shared_ptr_emplace<sf::Texture, std::__1::allocator<sf::Texture> >::~__shared_ptr_emplace() in main.cpp.o
      std::__1::__shared_ptr_emplace<sf::Texture, std::__1::allocator<sf::Texture> >::__on_zero_shared() in main.cpp.o
  "sf::Transform::scale(float, float)", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
  "sf::Transform::translate(float, float)", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
  "sf::VideoMode::VideoMode(unsigned int, unsigned int, unsigned int)", referenced from:
      _main in main.cpp.o
  "sf::Time::asMicroseconds() const", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
  "sf::Clock::getElapsedTime() const", referenced from:
      DisplayManager::draw() in display_manager.cpp.o
  "sf::Image::getSize() const", referenced from:
      _main in main.cpp.o
  "sf::Image::getPixel(unsigned int, unsigned int) const", referenced from:
      _main in main.cpp.o
  "sf::Window::isOpen() const", referenced from:
      _main in main.cpp.o
  "vtable for sf::VertexArray", referenced from:
      DisplayManager::DisplayManager(sf::RenderTarget&, sf::RenderWindow&, World&, Colony&) in display_manager.cpp.o
      _main in main.cpp.o
      World::World(unsigned int, unsigned int) in main.cpp.o
      Colony::Colony(float, float, unsigned int) in main.cpp.o
      World::~World() in main.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [AntSimulator] Error 1
make[1]: *** [CMakeFiles/AntSimulator.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by duocang** [comment]
> @duocang thanks for your help! ~I think I found the solution~.
> 
> Previously, I compile SFML with the source code downloaded from the official website.
> However, I found the released source code is 2.5.1 which released in 2018.
> It surely does not support M1 chip although it allows you to compile for x86_64 and install.
> 
> I read the commits of their repository and found they support M1 chip since [SFML/SFML@252d26a](https://github.com/SFML/SFML/commit/252d26afb0aa6375e08ef95b754cccf60f31b6df).
> 
> I compiled SFML with `-DCMAKE_OSX_ARCHITECTURES=arm64`.
> 
> Finally, I can compile AntSimulator without any error :)
> 
> But, it still cannot run with the following error,
> 
> ```
> ./AntSimulator 
> dyld: Library not loaded: @rpath/../Frameworks/freetype.framework/Versions/A/freetype
>   Referenced from: /usr/local/sfml-graphics.framework/Versions/2.5.1/sfml-graphics
>   Reason: no suitable image found.  Did find:
>         /Library/Frameworks/freetype.framework/Versions/A/freetype: no matching architecture in universal wrapper
>         /Library/Frameworks/freetype.framework/Versions/A/freetype: no matching architecture in universal wrapper
> [1]    69888 abort      ./AntSimulator
> ```
> 
> It still looks like a problem of SFML instead.

@pancak3  Can you please try`brew install freetype`?

**Comment by pancak3** [comment]
> > @duocang thanks for your help! ~I think I found the solution~.
> > Previously, I compile SFML with the source code downloaded from the official website.
> > However, I found the released source code is 2.5.1 which released in 2018.
> > It surely does not support M1 chip although it allows you to compile for x86_64 and install.
> > I read the commits of their repository and found they support M1 chip since [SFML/SFML@252d26a](https://github.com/SFML/SFML/commit/252d26afb0aa6375e08ef95b754cccf60f31b6df).
> > I compiled SFML with `-DCMAKE_OSX_ARCHITECTURES=arm64`.
> > Finally, I can compile AntSimulator without any error :)
> > But, it still cannot run with the following error,
> > ```
> > ./AntSimulator 
> > dyld: Library not loaded: @rpath/../Frameworks/freetype.framework/Versions/A/freetype
> >   Referenced from: /usr/local/sfml-graphics.framework/Versions/2.5.1/sfml-graphics
> >   Reason: no suitable image found.  Did find:
> >         /Library/Frameworks/freetype.framework/Versions/A/freetype: no matching architecture in universal wrapper
> >         /Library/Frameworks/freetype.framework/Versions/A/freetype: no matching architecture in universal wrapper
> > [1]    69888 abort      ./AntSimulator
> > ```
> > 
> > 
> > It still looks like a problem of SFML instead.
> 
> @pancak3 Can you please try`brew install freetype`?

Still :(

**Comment by paulotex** [comment]
Hi:
This is with 11.2 on a Intel Mac.

I was able to compile and run with the following requisits:
* Xcode
* brew

With them installed, and with brew properly updated (`brew update and brew upgrade`), I did:
```
brew install sfml
git clone https://github.com/johnBuffer/AntSimulator.git
cd AntSimulator
mkdir build
cd build
CMAKE_PREFIX_PATH=/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers/ ccmake ../
```

That crazy long CMAKE_PREFIX_PATH is so that Cmake finds the system's OpenGL libs and headers.
